libdnf/hy-package.cpp

```diff
@@ -1053,6 +1053,7 @@ dnf_package_get_advisories(DnfPackage *pkg, int cmp_type)
     dataiterator_init(&di, pool, 0, 0, UPDATE_COLLECTION_NAME,
```

Comment:
What about to iterate over all advisories and then call advisory.getApplicablePackages(). Then search if advisory contains proper package?

Reply:
```
    dataiterator_init(&di, pool, 0, 0, 0, 0, 0);
    dataiterator_prepend_keyname(&di, UPDATE_COLLECTION);
    while (dataiterator_step(&di)) {
        dataiterator_setpos_parent(&di);
        Advisory advisory(sack, di.solvid);
        auto advisoryPkgs = advisory.getApplicablePackages()
```
Then iterate over packages and test them?

Reply:
I think that is a trade off between code readability and performance. It would be more understandable and simpler but a lot more objects would be created, it would make objects for all advisories and almost all their packages, this way we don't do any of that.

I have tested it with rhel appstream metadata, looping over all available packages and getting their advisories (without any modules setup). It went from ~40sec with the current implementation to ~3m30sec when constructing all the objects. (also I have noticed the `dnf_package_get_advisories` is not actually used in dnf or libdnf).

Even still I would prefer to use the faster current version.

Reply:
Good point